rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -55,252 +49,105 @@ class Airdrop(NamedTuple):
+            )
+        if airdrops_metadata_cache is not None:
+            response = jsonloads_dict(airdrops_metadata_cache)  # get cached response
+
+    if response is None:  # query fresh data
+        try:
+            response = requests.get(url=AIRDROPS_INDEX, timeout=CachedSettings().get_timeout_tuple()).json()  # noqa: E501
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Airdrops Index request failed due to {e!s}') from e
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.AIRDROPS_METADATA,),
+                value=rlk_jsondumps(response),
+            )
+
+    airdrops: dict[str, Airdrop] = {}
+    try:
+        poap_airdrops: dict[str, list[str]] = response['poap_airdrops']
+        for protocol_name, airdrop_data in response['airdrops'].items():
+            if 'new_asset_data' in airdrop_data:
+                try:
+                    new_asset_type = AssetType.deserialize(airdrop_data['new_asset_data']['asset_type'])  # noqa: E501
+                    if new_asset_type == AssetType.EVM_TOKEN:
+                        crytpo_asset: CryptoAsset = get_or_create_evm_token(
+                            userdb=database,
+                            evm_address=airdrop_data['new_asset_data']['address'],
+                            chain_id=ChainID.deserialize(airdrop_data['new_asset_data']['chain_id']),
+                            decimals=airdrop_data['new_asset_data']['decimals'],
+                            name=airdrop_data['new_asset_data']['name'],
+                            symbol=airdrop_data['new_asset_data']['symbol'],
+                        )
+                    else:
+                        asset = Asset(airdrop_data['asset_identifier'])
+                        if asset.exists() is True:
+                            crytpo_asset = asset.resolve_to_crypto_asset()  # use the local values of user, if it pre-existed  # noqa: E501
+                        else:
+                            crytpo_asset = CryptoAsset.initialize(
+                                identifier=airdrop_data['asset_identifier'],
+                                asset_type=new_asset_type,
+                                name=airdrop_data['new_asset_data']['name'],
+                                symbol=airdrop_data['new_asset_data']['symbol'],
+                                coingecko=airdrop_data['new_asset_data'].get('coingecko', None),
+                                cryptocompare=airdrop_data['new_asset_data'].get('cryptocompare', None),  # noqa: E501
+                            )
+                            GlobalDBHandler.add_asset(crytpo_asset)
+                except DeserializationError as e:
+                    raise RemoteError(f'Airdrops Index contains an invalid ChainID or AssetType of a token. {e!s}') from e  # noqa: E501
```

Comment:
As I said the try/except blocks should be as small as required

```suggestion
                try:
                     new_asset_type = AssetType.deserialize(airdrop_data['new_asset_data']['asset_type'])  # noqa: E501
                     if (chain_id_raw := airdrop_data.get('new_asset_data', {}).get('chain_id')) is not None:
                          chain_id = ChainID.deserialize(chain_id_raw)
                except DeserializationError as e:
                    raise RemoteError(f'Airdrops Index contains an invalid ChainID or AssetType of a token. {e!s}') from e  # noqa: E501
```